includes/payment-tokens/class-wc-stripe-payment-tokens.php

```diff
@@ -301,44 +301,48 @@ public function woocommerce_get_customer_upe_payment_tokens( $tokens, $user_id,
 
 			// Retrieve the payment methods for the enabled reusable gateways.
 			$payment_methods = [];
+
+			$reusable_payment_method_types = array_keys( self::UPE_REUSABLE_GATEWAYS_BY_PAYMENT_METHOD );
+			$active_payment_method_types = [];
+
 			if ( $gateway->is_oc_enabled() ) {
-				// For OC, get all available payment method types
-				foreach ( self::UPE_REUSABLE_GATEWAYS_BY_PAYMENT_METHOD as $payment_method_type => $reausable_gateway_id ) {
+				// For Optimized Checkout, get all available payment method types.
+				foreach ( $reusable_payment_method_types as $payment_method_type ) {
 					$payment_method_instance = WC_Stripe_UPE_Payment_Gateway::get_payment_method_instance( $payment_method_type );
 					if ( $payment_method_instance ) {
-						$retrieved_methods = $customer->get_payment_methods( $payment_method_type );
-						if ( ! empty( $retrieved_methods ) ) {
-							$payment_methods[] = $retrieved_methods;
-						}
+						$active_payment_method_types[] = $payment_method_type;
 					}
 				}
```

Comment:
@wjrosa, can you comment on why we're fetching all possible saved payment methods when Optimized Checkout is enabled? If we have a set of enabled payment methods, I am not sure how we would suddenly end up with additional payment methods enabled. Should we not stick to the same logic as for the non-Optimized Checkout case?

Reply:
I've refactored the logic to rely on the payment method configuration regardless of Optimized Checkout configuration: 6c9eb2f3e1d522b0aa70c74a458f4b1e6618f6a4